File: api/src/main/java/com/mystore/manager/api/admin/payload/ProfilPayload.java
package com.mystore.manager.api.admin.payload;


import com.mystore.manager.api.common.payload.GlobalDataPayload;
import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import lombok.Data;

import java.util.List;

@Data
@JsonIgnoreProperties(ignoreUnknown = true)
public class ProfilPayload extends GlobalDataPayload {
    private List<PrivilegePayload> privileges;
}

File: ui-admin/src/app/backend/payloads/admin/profilpayload.ts
import { PrivilegePayload } from "./privilegepayload";

export class ProfilPayload {
    id: number;
    code: string;
    name: string;
    privileges: Array<PrivilegePayload>;
    
    constructor(){
        this.id = null;
        this.code = null;
        this.name = '';
    }
}